src/help/help.go

```diff
@@ -64,9 +68,123 @@ func help(topic string) string {
 	if f, present := m[topic]; present {
 		return helpFromBuildRule(f.FuncDef)
 	}
+
+	// Check plugins
+	if pluginHelp := helpForPlugin(topic); pluginHelp != "" {
+		return pluginHelp
+	}
+
+	return ""
+}
+
+// helpForPlugin returns some help text for a plugin
+func helpForPlugin(topic string) string {
+	config, err := core.ReadDefaultConfigFiles(nil)
+	if err != nil {
+		panic("Failed to read config")
+	}
+	if _, ok := config.Plugin[topic]; ok {
+		message := fmt.Sprintf("${BOLD_BLUE}%v${RESET} is a plugin defined in the ${GREEN}.plzconfig${RESET} file.\n", topic)
+
+		// This is if there's a subrepo specified for the plugin. If there isn't, we'll check for the subrepo in the plugins/
+		var subrepoName string
+		buildLabel := core.BuildLabel{Name: "all"}
+		if val, ok := config.Plugin[topic].ExtraValues["subrepo"]; ok {
+			subrepoName = val[0]
+			buildLabel.Subrepo = subrepoName
+		} else {
+			buildLabel.Subrepo = path.Join("plugins", topic)
+		}
+
+		state := newState()
+
+		// Parse the subrepo (Run reads the plugin config into config)
+		plz.Run([]core.BuildLabel{buildLabel}, nil, state, config, state.TargetArch)
+		subrepo := state.Graph.Subrepo(buildLabel.Subrepo)
+		if subrepo == nil {
+			log.Fatalf("Tried to get subrepo %v but failed", buildLabel.Subrepo)
+		}
+
+		return getPluginOptionsAndBuildDefs(subrepo, message)
+	}
 	return ""
 }
 
+// getPluginOptionsAndBuildDefs looks for information for the plugin specified in the config file
+func getPluginOptionsAndBuildDefs(subrepo *core.Subrepo, message string) string {
+	config := subrepo.State.Config
+	if config.PluginDefinition.Description != "" {
+		message += "\n" + config.PluginDefinition.Description + "\n"
+	}
+	if config.PluginDefinition.DocumentationSite != "" {
+		message += "\n" + config.PluginDefinition.DocumentationSite + "\n"
+	}
+	configOptions := ""
+	for _, v := range config.PluginConfig {
+		valueType := v.Type
+		if v.Type == "" {
+			valueType = "string"
+		}
+		var optional string
+		if v.Optional {
+			optional = " (optional) "
+		}
+		configOptions += fmt.Sprintf("${BLUE}   %v${RESET} ${GREEN}(%v)${RESET}${WHITE}%v${RESET}Default value: %v\n",
+			strings.ToLower(v.ConfigKey),
+			valueType,
+			optional,
+			v.DefaultValue)
+	}
+	if configOptions != "" {
+		message += "\n${BOLD_YELLOW}This plugin has the following options:${RESET}\n" + configOptions
+	}
+
+	buildFuncMap := map[string]*asp.Statement{}
+	populatePluginBuildFuncs(buildFuncMap, subrepo)
```

Comment:
Might be nicer to return `buildFuncMap` from `populatePluginBuildFuncs` since it's meant to be only mutated within it.